src/server/hook_func.c

```diff
@@ -6012,128 +6063,124 @@ add_pending_mom_allhooks_action(void *minfo, unsigned int action)
-	while (phook) {
-		add_pending_mom_hook_action((mominfo_t *)minfo,
-			phook->hook_name, action);
-		phook = (hook *)GET_NEXT(phook->hi_exechost_periodic_hooks);
-	}
-
-	phook = (hook *)GET_NEXT(svr_exechost_startup_hooks);
-	while (phook) {
-		add_pending_mom_hook_action((mominfo_t *)minfo,
-			phook->hook_name, action);
-		phook = (hook *)GET_NEXT(phook->hi_exechost_startup_hooks);
-	}
-
-	phook = (hook *)GET_NEXT(svr_execjob_launch_hooks);
+	phook = (hook *)GET_NEXT(svr_allhooks);
 	while (phook) {
-		add_pending_mom_hook_action((mominfo_t *)minfo,
-			phook->hook_name, action);
-		phook = (hook *)GET_NEXT(phook->hi_execjob_launch_hooks);
+		if (phook->hook_name &&	(phook->event & MOM_EVENTS)) {
+			add_pending_mom_hook_action((mominfo_t *)minfo,
+				phook->hook_name, action);
+		}
+		phook = (hook *)GET_NEXT(phook->hi_allhooks);
 	}
+}
 
-	phook = (hook *)GET_NEXT(svr_execjob_attach_hooks);
-	while (phook) {
-		add_pending_mom_hook_action((mominfo_t *)minfo,
-			phook->hook_name, action);
-		phook = (hook *)GET_NEXT(phook->hi_execjob_attach_hooks);
-	}
+/**
+ * @brief
+ *		clears out reply_expected flags of timed out actions
+ *		and delete their wait tasks
+ *
+ * @see
+ * 		handle_hook_sync_timeout
+ *
+ * @param[in]	tid	- transaction id of timedout hook sync sequence
+ * 
+ * @return void
+ */
+void
+clear_timed_out_reply_expected(long long int tid)
+{
+	int	 i, j;
+	mom_hook_action_t *pact;
+	struct work_task *ptask, *tmp_task;
+	mominfo_t *pmom;
+	struct def_hk_cmd_info *info;
 
-	phook = (hook *)GET_NEXT(svr_execjob_resize_hooks);
-	while (phook) {
-		add_pending_mom_hook_action((mominfo_t *)minfo,
-			phook->hook_name, action);
-		phook = (hook *)GET_NEXT(phook->hi_execjob_resize_hooks);
-	}
+	for (i = 0; i < mominfo_array_size; i++) {
 
-	phook = (hook *)GET_NEXT(svr_execjob_abort_hooks);
-	while (phook) {
-		add_pending_mom_hook_action((mominfo_t *)minfo,
-			phook->hook_name, action);
-		phook = (hook *)GET_NEXT(phook->hi_execjob_abort_hooks);
-	}
+		if ((pmom = mominfo_array[i]) == NULL)
+			continue;
 
-	phook = (hook *)GET_NEXT(svr_execjob_postsuspend_hooks);
-	while (phook) {
-		add_pending_mom_hook_action((mominfo_t *)minfo,
-			phook->hook_name, action);
-		phook = (hook *)GET_NEXT(phook->hi_execjob_postsuspend_hooks);
-	}
+		if (pmom->mi_num_action && pmom->mi_action) {
 
-	phook = (hook *)GET_NEXT(svr_execjob_preresume_hooks);
-	while (phook) {
-		add_pending_mom_hook_action((mominfo_t *)minfo,
-			phook->hook_name, action);
-		phook = (hook *)GET_NEXT(phook->hi_execjob_preresume_hooks);
+			for (j = 0; j < pmom->mi_num_action; j++) {
+				pact = pmom->mi_action[j];
+				if (pact && pact->reply_expected && (pact->tid == tid)) {
+					snprintf(log_buffer, sizeof(log_buffer),
+						"timedout, clearing reply_expected for %d event[%lld] of %s hook for %s",
+						pact->reply_expected, tid, pact->hookname, pmom->mi_host);
+					log_event(PBSEVENT_DEBUG3, PBS_EVENTCLASS_SERVER,
```

Comment:
log_eventf?

Reply:
Done here and one more place down somewhere. https://github.com/openpbs/openpbs/pull/1992/commits/d54ce17afb4e9d6cd1a3da070504726722e60e5c